layers/cmd/valueset.go

```diff
@@ -211,7 +216,36 @@ func (vsets Valuesets) Lookup(ctx *ls.Context, req ls.ValuesetLookupRequest) (ls
 		ctx.GetLogger().Debug(map[string]interface{}{"valueset.found": found})
 		return found, nil
 	}
-	for _, id := range req.TableIDs {
+	var n int = len(req.TableIDs)
+	var wg sync.WaitGroup
+	wg.Add(n)
+	results := make([]ls.ValuesetLookupResponse, n)
+	errs := make([]error, n)
+	for idx, id := range req.TableIDs {
+		go func(i int) {
```

Comment:
Add id:=id before goroutine. You are using id in the goroutine, which may change